examples/long_running_subscriber/main.go

```diff
@@ -0,0 +1,349 @@
+	flag.Parse()
+
+	if *token == "" {
+		log.Fatal("Usage: go run main.go -token=<token> [-processors=<count>]")
+	}
+
+	// Create context that cancels on interrupt
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Handle graceful shutdown
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// Create Sailhouse client
+	client := sailhouse.NewSailhouseClient(*token)
+
+	// Create processing stats
+	stats := &ProcessingStats{}
+
+	// Create subscriber with custom options
+	options := &sailhouse.SubscriberOptions{
+		PerSubscriptionProcessors: *processors,
+		PollInterval:              500 * time.Millisecond,
+		MaxRetries:                3,
+		RetryDelay:                2 * time.Second,
+		ErrorHandler: func(err error) {
+			stats.processingErrors.Add(1)
+			log.Printf("Processing error: %v", err)
+		},
+	}
+
+	subscriber := client.NewSubscriber(options)
+
+	// Register subscription handlers
+	registerHandlers(subscriber, stats)
+
+	// Start the subscriber
+	fmt.Printf("Starting subscriber with %d processors per subscription...\n", *processors)
+	err := subscriber.Start(ctx)
+	if err != nil {
+		log.Fatalf("Failed to start subscriber: %v", err)
+	}
+
+	fmt.Println("Subscriber started successfully!")
+	fmt.Println("Registered subscriptions:")
+	for _, sub := range subscriber.GetSubscriptions() {
+		fmt.Printf("  - %s/%s\n", sub.Topic, sub.Subscription)
+	}
+	fmt.Println()
+
+	// Start stats reporting
+	go reportStats(ctx, stats)
+
+	// Wait for shutdown signal
+	select {
+	case sig := <-sigChan:
+		fmt.Printf("\nReceived signal %v, shutting down gracefully...\n", sig)
+	case <-ctx.Done():
+		fmt.Println("\nContext cancelled, shutting down...")
+	}
+
+	// Graceful shutdown
+	fmt.Println("Stopping subscriber...")
+	subscriber.Stop()
+	fmt.Println("Subscriber stopped.")
+	fmt.Printf("Final stats: %s\n", stats)
+}
+
+func registerHandlers(subscriber *sailhouse.SailhouseSubscriber, stats *ProcessingStats) {
+	// Handler for user registration events
+	subscriber.Subscribe("user-registrations", "registration-processor", func(ctx context.Context, event *sailhouse.Event) error {
+		stats.eventsProcessed.Add(1)
+
+		fmt.Printf("[REGISTRATION] Processing event %s\n", event.ID)
+
+		var eventData map[string]interface{}
+		err := event.As(&eventData)
+		if err != nil {
+			stats.handlerErrors.Add(1)
+			return fmt.Errorf("failed to parse event data: %w", err)
+		}
+
+		eventType, ok := eventData["type"].(string)
+		if !ok {
+			stats.handlerErrors.Add(1)
+			return fmt.Errorf("missing or invalid event type")
+		}
+
+		switch eventType {
+		case "user.signup":
+			return handleUserSignup(eventData, event.Metadata)
+		case "user.email_verified":
+			return handleEmailVerified(eventData, event.Metadata)
+		case "user.trial_started":
+			return handleTrialStarted(eventData, event.Metadata)
+		default:
+			fmt.Printf("[REGISTRATION] Unknown event type: %s\n", eventType)
+			return nil
+		}
```

Comment:
I think this is in all of the examples here - worth revisiting I think (or maybe just not adding for now)

Reply:
Revisted